controllers/btpoperator_controller_test.go

```diff
@@ -249,200 +248,314 @@ var _ = Describe("BTP Operator controller", Ordered, func() {
 
-			simulateRestart(ctx, cr)
+			err := os.RemoveAll(chartUpdatePath)
+			Expect(err).To(BeNil())
+			err = os.RemoveAll(resourcesUpdatePath)
+			Expect(err).To(BeNil())
 
-			Expect(pullFromBtpManagerConfigMap(oldChartVersionKey)).To(Equal(initChartVersion))
-			Expect(pullFromBtpManagerConfigMap(currentCharVersionKey)).To(Equal(initChartVersion))
+			ChartPath = defaultChartPath
+			ResourcesPath = defaultResourcesPath
 		})
 
-		When("update of all resources names and bump chart version", Label("test-update"), func() {
-			It("new resources (with new name) should be created and old ones removed", func() {
-				err := ymlutils.TransformCharts(updatePath, suffix)
+		When("update all resources names and bump chart version", Label("test-update"), func() {
+			It("new resources (with new names) should be created and old ones removed", func() {
+				err := ymlutils.CopyManifestsFromYamlsIntoOneYaml(getApplyPath(), getToDeleteYamlPath())
 				Expect(err).To(BeNil())
 
-				err = ymlutils.UpdateVersion(updatePath, newChartVersion)
+				err = ymlutils.AddSuffixToNameInManifests(getApplyPath(), suffix)
 				Expect(err).To(BeNil())
 
-				simulateRestart(ctx, cr)
-
-				Expect(pullFromBtpManagerConfigMap(oldChartVersionKey)).To(Equal(initChartVersion))
-				Expect(pullFromBtpManagerConfigMap(currentCharVersionKey)).To(Equal(newChartVersion))
+				err = ymlutils.UpdateChartVersion(chartUpdatePath, newChartVersion)
+				Expect(err).To(BeNil())
 
-				oldCount, newCount := countResources()
-				Expect(oldCount).To(BeZero())
-				Expect(newCount >= minimalExpectedElementsCount).To(BeTrue())
+				Eventually(reconciler.workqueueSize).WithTimeout(time.Second * 5).WithPolling(time.Millisecond * 100).Should(Equal(0))
+				_, err = reconciler.Reconcile(ctx, controllerruntime.Request{NamespacedName: apimachienerytypes.NamespacedName{
+					Namespace: cr.Namespace,
+					Name:      cr.Name,
+				}})
+				Expect(err).To(BeNil())
 
-				Eventually(updateCh).Should(Receive(matchReadyCondition(types.StateReady, metav1.ConditionTrue, UpdateDone)))
+				actualNumOfOldResources, err := countResourcesForGivenChartVer(gvks, initChartVersion)
+				Expect(err).To(BeNil())
+				Expect(actualNumOfOldResources).To(Equal(0))
+				actualNumOfNewResources, err := countResourcesForGivenChartVer(gvks, newChartVersion)
+				Expect(err).To(BeNil())
+				Expect(actualNumOfNewResources).To(Equal(initResourcesNum))
 			})
 		})
 
-		When("update of all resources names and leave same chart version", Label("test-update"), func() {
-			It("new ones not created because version is not changed, old ones should stay", Label("test-update"), func() {
-				err := ymlutils.TransformCharts(updatePath, suffix)
+		When("update some resources names and bump chart version", Label("test-update"), func() {
+			It("resources with new names should be created, resources with old names should be deleted, unchanged resources should stay and receive new chart version", func() {
```

Comment:
So essentially all resources (if they stay) should have the chart version updated, should not they?

Reply:
correct

Reply:
changed the message to `all applied resources should receive new chart version, resources with new names should replace the ones with old names`